packages/protocol/contracts/L1/hooks/AssignmentHook.sol

```diff
@@ -120,7 +127,9 @@ contract AssignmentHook is EssentialContract, IHook {
                 refund = msg.value - input.tip;
             }
             // Paying ERC20 tokens
-            IERC20(assignment.feeToken).safeTransferFrom(msg.sender, blk.assignedProver, proverFee);
+            IERC20(assignment.feeToken).safeTransferFrom(
+                meta.coinbase, blk.assignedProver, proverFee
```

Comment:
fix bug in ERC20 proverFee payment (previous from TaikoL1, now from L2 block's coinbase address)